update/sync/syncHeadersByHash_test.go

```diff
@@ -0,0 +1,179 @@
+package sync
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/update"
+	"github.com/ElrondNetwork/elrond-go/update/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewMissingheadersByHashSyncer_NilParamsShouldErr(t *testing.T) {
+	t.Parallel()
+
+	okArgs := getMisingHeadersByHashSyncerArgs()
+
+	testInput := make(map[ArgsNewMissingHeadersByHashSyncer]error)
+
+	nilStorerArgs := okArgs
+	nilStorerArgs.Storage = nil
+	testInput[nilStorerArgs] = dataRetriever.ErrNilHeadersStorage
+
+	nilCacheArgs := okArgs
+	nilCacheArgs.Cache = nil
+	testInput[nilCacheArgs] = update.ErrNilCacher
+
+	nilMarshalizerArgs := okArgs
+	nilMarshalizerArgs.Marshalizer = nil
+	testInput[nilMarshalizerArgs] = dataRetriever.ErrNilMarshalizer
+
+	nilRequestHandlerArgs := okArgs
+	nilRequestHandlerArgs.RequestHandler = nil
+	testInput[nilRequestHandlerArgs] = update.ErrNilRequestHandler
+
+	for args, expectedErr := range testInput {
+		mhhs, err := NewMissingheadersByHashSyncer(args)
+		require.True(t, check.IfNil(mhhs))
+		require.Equal(t, expectedErr, err)
+	}
+}
+
+func TestNewMissingheadersByHashSyncer_OkValsShouldWork(t *testing.T) {
+	t.Parallel()
+
+	args := getMisingHeadersByHashSyncerArgs()
+	mhhs, err := NewMissingheadersByHashSyncer(args)
+	require.NoError(t, err)
+	require.NotNil(t, mhhs)
+}
+
+func TestSyncHeadersByHash_SyncMissingHeadersByHash_HeaderInCacheShouldWork(t *testing.T) {
```

Comment:
Also here, remove the last _ from function names

Reply:
done